### ISSUE TITLE
Update linter to check jsx files

### ISSUE THREAD

**Issue by jfly** [cubing/ccm#312]
As we move to react, this is going to become more and more important.

http://www.sitepoint.com/comparison-javascript-linting-tools/ and https://github.com/STRML/JSXHint say we should be using [ESLint](http://eslint.org/).

IMO, this is important to do before we start heavy React development.

**Comment by coder13** [comment]
I agree on eslint as I hate having to keep turning it off. I can get this setup if you want.

**Comment by jfly** [comment]
If you've got the time to work on this, please assign this to yourself and go for it!

**Comment by coder13** [comment]
I think I fixed this issue during my big rewrite, I'll close this after I merge to master.
